contracts/AllocationLedger.sol

```diff
@@ -137,49 +170,92 @@ contract AllocationLedger is ReentrancyGuard, Context, Ownable, Pausable {
         );
     }
 
+    /**
+     * @dev Used to claim available rewards to the `_msgSender()` address.
+     */
+    function claimRewards()
+        external
+        whenPausedAction(PAUSE_ACTION_DEPOSIT)
+        whenNotPausedAction(PAUSE_ACTION_CLAIM)
+        onlyWhitelisted(_msgSender())
+    {
+        require(deposits[_msgSender()] > 0, "Account never deposited");
+
+        uint256 accountShare = getAccountRewards(_msgSender());
+        uint256 alreadyClaimed = claims[_msgSender()];
+        uint256 amount = accountShare - alreadyClaimed;
+
+        require(amount > 0, "Nothing to claim");
+
+        rewardsToken.transfer(_msgSender(), amount);
+        claims[_msgSender()] = alreadyClaimed.add(amount);
```

Comment:
claims[_msgSender()] = alreadyClaimed.add(amount);
 totalRewardsClaimed += amount;

Before the actual transfer